.github/workflows/push_pr.yml

```diff
@@ -0,0 +1,36 @@
+name: Matrix Testing
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 1'
```

Comment:
Why would this need to run when there are no new commits?

Reply:
because dependencies may change and break something.